### PR TITLE
Adding FileLogger object. #7229.

### PR DESCRIPTION
**Pull request by cropalato**
https://dev.icinga.org/issues/7229

### CODE REVIEW

File: manifests/object/filelogger.pp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# See the inline comments.
+#
+
+define icinga2::object::filelogger (
+  $ensure                    = 'file',
+  $object_name               = $name,
+  $path                      = undef,
+  $severity                  = undef,
+  $target_dir                = '/etc/icinga2/conf.d',
+  $target_file_name          = "${object_name}.conf",
```
Comment (nickchappell) [inline manifests/object/filelogger.pp]: You'll have to set `$target_file_name` to `$name`. The `$name` variable is available and defined when Puppet evaluates the defined type, but it looks like the `$object_name` variable doesn't get 'set' internally until after it tries to create the file resource. When I tested it, it created the file as `/etc/icinga2/conf.d/.conf`.

**Comment by cropalato** [comment]
fixed

**Comment by nickchappell** [comment]
Just wanted to give you a heads up that I'll be traveling with family for a little while and may not able to review/merge this until after Christmas.

**Comment by nickchappell** [comment]
I merged this into the develop branch. Thanks @cropalato!

**Pull request closed without merging**